packages/hw-transport-node-hid-singleton/src/TransportNodeHid.ts

```diff
@@ -12,6 +12,22 @@ import { identifyUSBProductId } from "@ledgerhq/devices";
 import { CantOpenDevice } from "@ledgerhq/errors";
 import { listenDevices } from "./listenDevices";
 let transportInstance;
+
+const DISCONNECT_TIMEOUT = 5000;
+let disconnectTimeout;
+const clearDisconnectTimeout = () => {
+  if (disconnectTimeout) {
+    clearTimeout(disconnectTimeout);
+  }
+};
+
+const setDisconnectTimeout = () => {
+  disconnectTimeout = setTimeout(
```

Comment:
shouldn't it implicitly clearDisconnectTimeout here in case it was called twice?

Reply:
You're probably right, I'll check :beau: